sig-scalability/slos/slos.md

```diff
@@ -92,17 +92,53 @@ of the cases.
 We are looking into extending the set of SLIs/SLOs to cover more parts of
 Kubernetes.
 
+```
+Prerequisite: Kubernetes cluster is available and serving.
+```
+
 ### Steady state SLIs/SLOs
 
 | Status | SLI | SLO | User stories, test scenarios, ... |
 | --- | --- | --- | --- |
+| __Official__ | Latency<sup>[1](#footnote1)</sup> of mutating<sup>[2](#footnote2)</sup> API calls for single objects for every (resource, verb) pair, measured as 99th percentile over last 5 minutes | In default Kubernetes installation, for every (resource, verb) pair, excluding virtual and aggregated resources and Custom Resource Definitions, 99th percentile per cluster-day<sup>[3](#footnote3)</sup> <= 1s | [Details](./api_call_latency.md) |
+| __Official__ | Latency<sup>[1](#footnote1)</sup> of non-streaming read-only<sup>[4](#footnote3)</sup> API calls for every (resource, scope<sup>[5](#footnote4)</sup>) pair, measured as 99th percentile over last 5 minutes | In default Kubernetes installation, for every (resource, scope) pair, excluding virtual and aggregated resources and Custom Resource Definitions, 99th percentile per cluster-day (a) <= 1s if `scope=resource` (b) <= 5s if `scope=namespace` (c) <= 30s if `scope=cluster` | [Details](./api_call_latency.md) |
+
+<a name="footnote1">\[1\]</a>By latency of API call in this doc we mean time
+from the moment when apiserver gets the request to last byte of response sent
+to the user.
+
+<a name="footnote2">\[2\]</a>By mutating API calls we mean POST, PUT, DELETE
+and PATCH.
+
+<a name="footnote3">\[3\]</a> For the purpose of visualization it will be a
+sliding window. However, for the purpose of reporting the SLO, it means one
+point per day (whether SLO was satisfied on a given day or not).
+
+<a name="footnote4">\[4\]</a>By non-streaming read-only API calls we mean GET
+requests without `watch=true` option set. (Note that in Kubernetes internally
+it translates to both GET and LIST calls).
+
+<a name="footnote5">\[5\]</a>A scope of a request can be either (a) `resource`
+if the request is about a single object, (b) `namespace` if it is about objects
+from a single namespace or (c) `cluster` if it spawns objects from multiple
```

Comment:
This might have already been discussed earlier - but it seems like non-namespaced resources (like nodes) will be put under 'cluster' category? If so, it means we give as much as 30s for 'LIST nodes' calls for e.g. Are we ok with that?

Reply:
I think it's what we should do. For things like PVs it will be possible to have a looot of them.

Reply:
I partly agree with you - but I don't have any data to prove how they compare with the order of namespaced resources :)

One idea is to have a different bound for non-namespaced category (10s) which will for e.g align well with our current latency for "list nodes" that's ~5s on 5k. It'll also help us to not link this category with other existing ones. But I'll leave the decision here to you.

Reply:
Ok, actually that value seems to be <2s without those slow calls coming from cpu-starved heapster nanny.

Reply:
I would say that we may consider tuning this in the past, but as long as we don't have cluster loader and won't measure exactly what is here (5-min window, etc.), let's not tune it before then.

Reply:
Assuming you meant 'future' by 'past', that sounds reasonable to me :)
Maybe add a TODO for revisiting the SLO for non-namespaced resources?

Reply:
Yes - sorry, I meant future :)

Sure - adding TODO sounds good to me - adding now.

Reply:
Added a TODO.

Reply:
Lgtm, thx.